File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.java
class Solution {
    public int maxVowels(String s, int k) {
        //Vowel List
        Set<Character> vowels = new HashSet<>(Arrays.asList('a', 'e', 'i', 'o', 'u'));

        int currentVowels = 0;
        //Fill window
        for(int i= 0; i < k; i++){
            if(vowels.contains(s.charAt(i))){
                currentVowels++;
            }
        }

        int maxVowels = currentVowels;
        //Slide Window
        for(int i = k; i < s.length(); i++){

      

            char charIn = s.charAt(i);
            char charOut = s.charAt(i - k);

            if(vowels.contains(charIn)){
                currentVowels++;
            }
            
            if(vowels.contains(charOut)){
                currentVowels--;
            }

            maxVowels = Math.max(maxVowels, currentVowels);
        }
        return maxVowels;
    }
}
    

File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        #Initialize which letters are vowels
        vowels = ['a','e','i','o','u']
        currentVowels = 0

        #Iterate through array
        
        for i in range(k):
            if s[i] in vowels:
                currentVowels += 1
        
        maxVowels = currentVowels

        for i in range(k, len(s)):
            charIn = s[i]       
            charOut = s[i - k]  
            
            if charIn in vowels:
                currentVowels += 1
            if charOut in vowels: 
                currentVowels -= 1

            # 3. Actualizar el máximo
            maxVowels = max(maxVowels, currentVowels)

        return maxVowels

    
        